### PR TITLE
fix(travis): add check to verify existence of udev

### PR DESCRIPTION
**Pull request by mittachaitu**
Signed-off-by: mittachaitu <sai.chaithanya@mayadata.io>

**What this PR does / why we need it**:
This PR adds a test case to verify the existence of udev inside cstor-pool container.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
- After merging https://github.com/openebs/openebs/pull/2943 Travis of this PR needs to trigger once again. 

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] Labelled this PR & related issue with `documentation` tag
- [ ] PR messages has document related information
- [ ] Labelled this PR & related issue with `breaking-changes` tag
- [ ] PR messages has breaking changes related information
- [ ] Labelled this PR & related issue with `requires-upgrade` tag
- [ ] PR messages has upgrade related information
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -19,6 +19,12 @@ addons:
   apt:
     update: true
 
+before_install:
+  - sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
+  - sudo apt-get update -qq
+  - sudo apt-get install --yes -qq gcc-6
```
Comment (pawanpraka1) [inline .travis.yml]: no need to have gcc-6, installed gcc version is sufficient.

File: .travis.yml [hunk 2]
```diff
@@ -19,6 +19,12 @@ addons:
...
+  - sudo apt-get install --yes -qq libudev-dev
+  - sudo unlink /usr/bin/gcc && sudo ln -s /usr/bin/gcc-6 /usr/bin/gcc
```
Comment (pawanpraka1) [inline .travis.yml]: not needed

File: ci/travis-ci.sh [hunk 3]
```diff
@@ -23,7 +23,12 @@ export MAYACTL="$GOPATH/src/github.com/openebs/maya/bin/maya/mayactl"
 ./ci/build-maya.sh
 rc=$?; if [[ $rc != 0 ]]; then exit $rc; fi
 
-curl https://raw.githubusercontent.com/openebs/openebs/master/k8s/ci/test-script.sh > test-script.sh
+#curl https://raw.githubusercontent.com/openebs/openebs/master/k8s/ci/test-script.sh > test-script.sh
+curl https://raw.githubusercontent.com/openebs/openebs/fb0100f6a2f343b3c77e52f922dd40f0e7ded417/k8s/ci/test-script.sh > test-script.sh
```
Comment (pawanpraka1) [inline ci/travis-ci.sh]: why this change? Why we are picking the file from a particular commit?

File: ci/udev_check.c [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+                fprintf(stderr, "device is not provided\n");
+                return (-1);
+        }
+
+        dev = udev_device_new_from_subsystem_sysname(udev, "block", device);
+
+        if ((dev != NULL) && udev_device_get_is_initialized(dev)) {
+                printf("device = %s is initialized by udev\n", device);
+        } else {
+                printf("device = %s is not initialized by udev\n", device);
```
Comment (pawanpraka1) [inline ci/udev_check.c]: modify this code and return non zero in this case, and use the return code to check for failure/success of this command.

File: ci/travis-ci.sh (hunk 3 above)
Comment (mittachaitu) [reply ci/travis-ci.sh]: Just to make sure that Travis should pass with the test case added. It will be reverted back once got approvals.

File: ci/udev_check.c (hunk 4 above)
Comment (mittachaitu) [reply ci/udev_check.c]: Since we are executing this via `kubectl exec -it ...` since the process execution i.e udev_check sda is executed successfully it returns 0 even if the process returns -1.

Comment (pawanpraka1) [reply ci/udev_check.c]: @mittachaitu
```
 $ kubectl exec -it openebs-ndm-dzzpk -n openebs -- ls abc
ls: cannot access 'abc': No such file or directory
command terminated with exit code 2
k8s@e2e1-master:~$ echo $?
2
```

File: ci/udev_check.c [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        if (argc > 1) {
+                device = argv[1];
+        } else {
+                fprintf(stderr, "device is not provided\n");
+                return (-1);
+        }
+
+        dev = udev_device_new_from_subsystem_sysname(udev, "block", device);
+
+        if ((dev != NULL) && udev_device_get_is_initialized(dev)) {
```
Comment (mynktl) [inline ci/udev_check.c]: We need to handle failure case for `udev_device_get_is_initialized`.

File: ci/udev_check.c (hunk 4 above)
Comment (mynktl) [reply ci/udev_check.c]: let's print `errno` also

Comment (mittachaitu) [reply ci/udev_check.c]: > @mittachaitu
> 
> ```
>  $ kubectl exec -it openebs-ndm-dzzpk -n openebs -- ls abc
> ls: cannot access 'abc': No such file or directory
> command terminated with exit code 2
> k8s@e2e1-master:~$ echo $?
> 2
> ```

I might try in older versions of Kubernetes where it returned 0 .

File: .travis.yml (hunk 1 above)
Comment (mittachaitu) [reply .travis.yml]: Done

File: .travis.yml (hunk 2 above)
Comment (mittachaitu) [reply .travis.yml]: Removed

File: ci/udev_check.c (hunk 5 above)
Comment (mittachaitu) [reply ci/udev_check.c]: Handled

File: ci/udev_check.c (hunk 4 above)
Comment (mittachaitu) [reply ci/udev_check.c]: Done

File: ci/udev_check.c [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+int main(int argc, char **argv)
+{
+	char *device;
+	struct udev *udev;
+	struct udev_device *dev = NULL;
+	int initialized = -1;
+	int rc = -1;
+
+	if ((udev = udev_new()) == NULL) {
+	        printf("failed to get udev device\n");
```
Comment (pawanpraka1) [inline ci/udev_check.c]: print this also to stderr. fprintf(stderr,
